core/src/main/java/org/elasticsearch/transport/netty/NettyTransportChannel.java

```diff
@@ -134,6 +141,14 @@ public void sendResponse(Throwable error) throws IOException {
         transportServiceAdapter.onResponseSent(requestId, action, error);
     }
 
+    private void close() {
```

Comment:
well, we don't really do ops on the `NettyTransportChannel` but rather it's underlying fields  which is similar in my mind to any clean up on close code. But I'm with you if you want to change this - I personally find release/unreserve even more confusing :)